translate/storage/tmx.py

```diff
@@ -18,13 +18,19 @@
 
 """module for parsing TMX translation memeory files"""
 
+import unicodedata
+
 from lxml import etree
 
 from translate import __version__
 from translate.misc.xml_helpers import setXMLlang
 from translate.storage import lisa
 
 
+def remove_control_characters(s):
+    return "".join(ch for ch in s if unicodedata.category(ch)[0] != "C")
```

Comment:
This removes way too much for example right-to-left/left-to-right marks are C as well and you definitely do not want these to be removed.

The actual code in libxml rejects just a few ASCII control chars, see https://github.com/GNOME/libxml2/blob/46837d47d59c7b8c9bd1d08a6a717a90a7f1ceb6/include/libxml/chvalid.h#L108-L110

Also, I think this code is better placed to `translate/misc/xml_helpers.py` because it is probably useful for other XML based formats as well.

Reply:
ok, makes sense. I think I'll finish work and then open a new PR with everything at the right spot.
Is it ok like this?

Reply:
Yes, sounds good.